crates/vm/wasm/src/vm.rs

```diff
@@ -28,14 +30,13 @@ const MAX_QUERY_DEPTH: usize = 3;
 
 #[derive(Clone)]
 pub struct WasmVm {
-    cache: Cache,
+    cache: Option<Cache>,
 }
 
 impl WasmVm {
     pub fn new(cache_capacity: usize) -> Self {
-        // TODO: handle the case where cache capacity is zero (which means not to use a cache)
         Self {
-            cache: Cache::new(NonZeroUsize::new(cache_capacity).unwrap()),
+            cache: NonZeroUsize::new(cache_capacity).map(Cache::new),
```

Comment:
It seems changes from #143 related to the cache are also included in this PR.

Probably because you created the branch off the #143 branch instead of main.

Can you somehow rebase it so that changes from #143 aren't included in this PR?